server/src-lib/Hasura/GraphQL/Resolve/Types.hs

```diff
@@ -33,7 +33,8 @@ data QueryCtx
   | QCSelectAgg !SelOpCtx
   | QCFuncQuery !FuncQOpCtx
   | QCFuncAggQuery !FuncQOpCtx
-  | QCActionFetch !ActionSelectOpContext
+  | QCAsyncActionFetch !ActionSelectOpContext
+  | QCAction !ActionExecutionContext
```

Comment:
You can use
```haskell
| QCAction !SyncActionExecutionContext
```
to avoid the pattern match pitfall of  `ActionExecutionAsync`.

I suggest renaming 
- `ActionExecutionContext` -> `ActionMutationExecutionContext`
- ` SyncActionExecutionContext` -> `ActionExecutionContext`

After renaming `ActionExecutionContext` can be shared between "action query" and "async action query".

Reply:
Done